src/Functions/FunctionsHashing.h

```diff
@@ -83,7 +83,12 @@ namespace impl
     {
         SipHashKey ret;
 
-        const auto * tuple = checkAndGetColumn<ColumnTuple>(key.column.get());
+        const ColumnTuple * tuple = nullptr;
+        const auto * column = key.column.get();
+        if (isColumnConst(*column))
+            tuple = checkAndGetColumnConstData<ColumnTuple>(column);
+        else
+            tuple = checkAndGetColumn<ColumnTuple>(column);
```

Comment:
When a column is non-const, taking the first element out of a full column is obviously wrong.

Let me close this PR and remove the keyed functions entirely.